Summatives/m3-summative/LINQ/Program.cs

```diff
@@ -0,0 +1,732 @@
+                                           OrdersCount = (customer.Orders.Any()) ? customer.Orders.Count() : 0
+                                       };
+
+            foreach (var customer in customersorderscount)
+            {
+                Console.WriteLine("==============================================================================");
+                Console.WriteLine(customer.CustomerID);
+                Console.WriteLine("\tOrders Count");
+                Console.WriteLine(customer.OrdersCount);
+                Console.WriteLine("==============================================================================");
+                Console.WriteLine();
+            }
+        }
+
+        /// <summary>
+        /// Print a distinct list of product categories and the count of the products they contain
+        /// </summary>
+        static void PrintDistinctCategoriesandProductsCount()
+        {
+            List<Product> products = DataLoader.LoadProducts();
+            var productscategories = (from product in products
+                                     group product by product.Category);
+
+            string line = "{0,-15} {1,-3}";
+            Console.WriteLine(line, "Category", "Products Count");
+            Console.WriteLine("==============================");
+
+            foreach (var category in productscategories)
+            {
+                Console.WriteLine(line, category.Key,category.Count());
+            }
+        }
+
+        /// <summary>
+        /// Print a distinct list of product categories and the total units in stock
+        /// </summary>
+        static void PrintDistinctCategoriesandUnitsInStockCount()
+        {
+            List<Product> products = DataLoader.LoadProducts();
+            var productscategories = (from product in products
+                                      group product by product.Category);
+
+            string line = "{0,-15} {1,-3}";
+            Console.WriteLine(line, "Category", "UnitsInStock Count");
+            Console.WriteLine("==================================");
+
+            foreach (var category in productscategories)
+            {
+                var UnitsInStockList = (from product in products
+                                        where product.Category == category.Key
+                                        select product.UnitsInStock);
+                Console.WriteLine(line, category.Key, UnitsInStockList.Sum());
+            }
+        }
+
+        /// <summary>
+        /// Print a distinct list of product categories and the lowest priced product in that category
+        /// </summary>
+        static void PrintDistinctCategoriesandLowestPricedProduct()
+        {
+            List<Product> products = DataLoader.LoadProducts();
+            var productscategories = (from product in products
+                                      group product by product.Category);            
```

Comment:
You can use a select here to grab the lowest price in the group with the group name